packages/backend-tasks/src/tasks/PluginTaskSchedulerImpl.ts

```diff
@@ -35,16 +40,25 @@ export class PluginTaskSchedulerImpl implements PluginTaskScheduler {
     const knex = await this.databaseFactory();
 
     const worker = new TaskWorker(task.id, task.fn, knex, this.logger);
-    await worker.start(
-      {
-        version: 1,
-        initialDelayDuration: task.initialDelay?.toISO(),
-        recurringAtMostEveryDuration: task.frequency.toISO(),
-        timeoutAfterDuration: task.timeout.toISO(),
-      },
-      {
-        signal: task.signal,
-      },
-    );
+
+    const settings: any = {
+      initialDelayDuration: task.initialDelay?.toISO(),
+      timeoutAfterDuration: task.timeout.toISO(),
+    };
+
+    if (task.frequency) {
+      settings.version = 1;
+      (settings as TaskSettingsV1).recurringAtMostEveryDuration =
+        task.frequency.toISO();
+    } else {
+      settings.version = 2;
+      if (task.cadence) {
+        (settings as TaskSettingsV2).cadence = task.cadence;
```

Comment:
Hmm. This made me think. I don't think we want to be in a position where we long-term support multiple versions - it's more of a short term thing, to support transitioning from one version to the next.

What I mean with this is, we should probably leave v1 completely unchanged, and instead make v2 support both formats. I think it would make sense to add `cadence` as a string to v2, and letting that string contain EITHER an iso period OR a cron string - and then the reading end is smart enough to handle that depending on what it gets in the string (like, whether it starts with a P or not or so.

Reply:
Yeah, my initial thought was to leave keep the V1 schema as much in tact as possible, and have V2 do a completely different format like this. Then in a later change (say a month or two) show up with another MR completely removing the V1 settings with support for frequency so we aren't supporting both long term.

Most of the work in the MR was frankly just ensuring we're supporting both versions :P